### PR TITLE
Change cluster preinstalled apps UI

### PR DESCRIPTION
**Pull request by gusevda**
Closes https://github.com/giantswarm/roadmap/issues/1277.

This PR changes how the preinstalled apps are displayed on the cluster details page. Previously we displayed some static information based on the Release CR, information like - release component name, version and icon. With the changes from this PR, preinstalled apps are displayed the same way as other (user installed) apps. Some additional changes were implemented to the `App` component to work well with preinstalled apps:
- `App` component UI was limited so it's not possible to modify preinstalled apps (upgrade, change configuration or uninstall);
- information about the upstream version of an app was added to the app details.

Before:
Screenshot 2022-08-18 at 15 19 05

After:
Screenshot 2022-08-18 at 15 18 24

Extra refactoring:
- cluster's `hasOptionalIngress` setting was removed since it's always `True` for all the releases that are currently in use for all the providers.

### CODE REVIEW

File: src/components/MAPI/apps/ClusterDetailReleaseApps.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  const clientFactory = useHttpClientFactory();
+  const auth = useAuthProvider();
+
+  const releaseClient = useRef(clientFactory());
+  const { canGet: canGetReleases } = usePermissionsForReleases(
+    provider,
+    'default'
+  );
+
+  const isReleasesSupportedByProvider = supportsReleases(provider);
```
Comment (kuosandys) [inline src/components/MAPI/apps/ClusterDetailReleaseApps.tsx:47]: Thanks for checking if releases are supported before fetching them! In the case that they aren't though, would it still make sense to render this component?

File: src/components/MAPI/apps/ClusterDetailAppListWidgetVersion.tsx [hunk 2]
```diff
@@ -22,11 +26,20 @@ interface IClusterDetailAppListWidgetVersionProps
...
 
 const ClusterDetailAppListWidgetVersion: React.FC<
   React.PropsWithChildren<IClusterDetailAppListWidgetVersionProps>
-> = ({ app, catalogNamespace, canListAppCatalogEntries, ...props }) => {
+> = ({
+  app,
+  catalogNamespace,
+  canListAppCatalogEntries,
+  displayUpgradableStatus = true,
+  displayUpstreamVersion = false,
```
Comment (kuosandys) [inline src/components/MAPI/apps/ClusterDetailAppListWidgetVersion.tsx:40]: Just for my understanding, is the main function of the `displayUpstreamVersion` prop for rendering a variant of this version widget?

File: src/components/MAPI/apps/ClusterDetailReleaseApps.tsx (hunk 1 above)
Comment (gusevda) [reply src/components/MAPI/apps/ClusterDetailReleaseApps.tsx:47]: Actually, this component will be used only on vintage. For CAPI providers there will be another component where we will fetch `default-apps-xxx` bundle. So this check is redundant probably.

**Review by kuosandys — APPROVED** [review]
Looks good to me 👍 and thanks for removing the unused optional ingress code! I just have the 2 questions above, but they shouldn't block merging this PR

File: src/components/MAPI/apps/ClusterDetailAppListWidgetVersion.tsx (hunk 2 above)
Comment (gusevda) [reply src/components/MAPI/apps/ClusterDetailAppListWidgetVersion.tsx:40]: This is correct

**Comment by marians** [comment]
Will give it a short test drive

**Review by marians — APPROVED** [review]
Tested, works great!

I noticed that the text `These apps and services are preinstalled on your cluster and managed by Giant Swarm.` is not exactly appropriate. I think it fits for an iteration where we not only listed apps. By now I think we should just simplify it to

`These apps are preinstalled on your cluster and managed by Giant Swarm.`

**Pull request merged**